velox/type/DecimalUtil.cpp

```diff
@@ -48,6 +48,34 @@ std::string formatDecimal(uint8_t scale, int128_t unscaledValue) {
   return fmt::format(
       "{}{}{}", isNegative ? "-" : "", integralPart, fractionString);
 }
+
+int128_t rescale(
+    const int128_t& unscaledValue,
+    const uint8_t fromScale,
+    const std::pair<uint8_t, uint8_t>& toPrecisionScale) {
+  auto rescaledValue = unscaledValue;
+  auto rf = toPrecisionScale.second - fromScale;
+  if (rf >= 0) {
+    rescaledValue = unscaledValue * DecimalUtil::kPowersOfTen[rf];
+  } else {
+    rf = -1 * rf;
+    rescaledValue = unscaledValue / DecimalUtil::kPowersOfTen[rf];
+    int128_t remainder = unscaledValue % DecimalUtil::kPowersOfTen[rf];
+    if (unscaledValue >= 0) {
+      if (remainder > DecimalUtil::kPowersOfTen[rf] / 2) {
+        rescaledValue++;
+      }
+    } else {
+      if (remainder < -DecimalUtil::kPowersOfTen[rf] / 2) {
+        rescaledValue--;
+      }
+    }
+  }
+  // Check overflowing
+  VELOX_CHECK_LE(
```

Comment:
`VELOX_DCHECK_LE`

Reply:
It should not be only a debug check. Replaced with an if/else block and setting a flag to indicate overflow case.